### PR TITLE
add CHAINER_BENCHMARK_BACKENDS to filter backend

### PR DESCRIPTION
**Pull request by kmaehashi**

### CODE REVIEW

File: benchmarks/utils/backend.py [hunk 1]
```diff
@@ -62,6 +71,11 @@ class declaration) decorator.
     ... class ConvolutionBenchmark(object):
     ...     def time_benchmark(self):
     ...         ...
+
+    You can temporarily limit the backend variation by setting list of
+    comma-separted backend names to CHAINER_BENCHMARK_BACKENDS environment
```
Comment (asi1024) [inline benchmarks/utils/backend.py]: Typo: separted -> separated

File: benchmarks/utils/backend.py [hunk 2]
```diff
@@ -25,6 +26,14 @@
     'cpu-ideep',
 ]
 
+_enabled_backend_modes = (
+    os.environ['CHAINER_BENCHMARK_BACKENDS'].split(',')
+    if 'CHAINER_BENCHMARK_BACKENDS' in os.environ
+    else _backend_modes
+)
+
+assert all([x in _backend_modes for x in _enabled_backend_modes])
```
Comment (asi1024) [inline benchmarks/utils/backend.py]: How about checking not setting the same mode twice?

File: benchmarks/utils/backend.py (hunk 1 above)
Comment (kmaehashi) [reply benchmarks/utils/backend.py]: Fixed.

File: benchmarks/utils/backend.py (hunk 2 above)
Comment (kmaehashi) [reply benchmarks/utils/backend.py]: I think it's better not to check it, as it is harmless. Sometimes flexibility is important for automation. For example I may want to do `CHAINER_BENCHMARK_BACKENDS=cpu,$ADDITIONAL_BACKENDS` to test against arbitrary backends while ensuring that ``cpu`` is always covered.

Comment (asi1024) [reply benchmarks/utils/backend.py]: OK, I see.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**